staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -125,14 +126,17 @@ type PriorityLevelConfigurationReference struct {
 }
 
 // PolicyRuleWithSubjects prescribes a test that applies to a request to an apiserver. The test considers the subject
-// making the request, the verb being requested, and the resource to be acted upon.
+// making the request, the verb being requested, and the resource or subresource or non-resource to be acted upon.
 type PolicyRuleWithSubjects struct {
 	// `subjects` is the list of normal user, serviceaccount, or group that this rule cares about.
 	// +optional
 	Subjects []Subject `json:"subjects,omitempty" protobuf:"bytes,1,opt,name=subject"`
-	// `rule` is the target verb, resource or the subresource the rule cares about. APIGroups, Resources, etc.
+	// `rule` is the target verb and most attributes of the resource or subresource or non-resource the rule cares about. This includes APIGroups, Resources, etc.  This excludes the namespace, if any, identifying the resource or subresource to act upon.  This exclusion is because the PolicyRule datatype is copied from rbac, which identifies the namespace in a different way.
 	// Required.
 	Rule PolicyRule `json:"rule" protobuf:"bytes,2,opt,name=rule"`
+	// `objectNamespaces` identifies the allowable namespaces, if any, of the resource or subresource that the rule matches.  For a non-resource or non-namespaced resource this list does not matter.  To match a resource or subresource regardless of its namespace, use NamespaceAll.  If NamespaceAll appears in the list then it may be the only entry in the list.
+	// +optional
+	ObjectNamespaces []string `json:"objectNamespaces" protobuf:"bytes,3,opt,name=objectNamespaces"`
```

Comment:
in line 133: protobuf:"bytes,1,opt,name=subject" it uses singular 'subject'

while here: protobuf:"bytes,3,opt,name=objectNamespaces", plural objectNamespaces is used.

better to be consistent, IMO

Reply:
Oh!  That looks like a bug in the protobuf for the `Subjects` field.  Note that the golang and JSON names for that field use the plural.  It is by far the dominant practice that the JSON and protobuf names are kept consistent with the golang name regarding singular/plural.  I will open a separate PR to fix the protobuf for `Subjects`.